covid19-etl/etl/vac_tracker.py

```diff
@@ -0,0 +1,184 @@
+        records to submit in `self.location_data` and `self.time_series_data`.
+        Ignores any records that are already in Sheepdog (relies on unique
+        `submitter_id` to check)
+
+        Args:
+            url (str): URL at which the CSV file is available
+        """
+        print("Getting data from {}".format(url))
+        with closing(requests.get(url, stream=True)) as r:
+            data = r.json()
+            try:
+                for treatment in data["result"]["pageContext"]["treatments"]:
+                    node = treatment["node"]
+                    clinical_trial = self.parse_node(node)
+                    self.clinical_trials.append(clinical_trial)
+            except ValueError as e:
+                print(f"ERROR: value error. Detail {e}")
+
+    def parse_node(self, node):
+        """
+        Converts an element of an JSON file to data we can submit via Sheepdog
+
+        Args:
+            node (dict): node data
+
+        Returns:
+            dict:
+                - clinical trial data, in a format ready to be submitted to Sheepdog
+        """
+        clinical_trial = {
+            "projects": [{"code": self.project_code}],
+            "type": "clinical_trials",
+        }
+
+        for key, value in node.items():
+            if key not in MAP_FIELDS:
+                continue
+            gen3_field = MAP_FIELDS.get(key)[0]
+            gen3_field_type = MAP_FIELDS.get(key)[1]
+            if type(value) != gen3_field_type:
+                print(
+                    f"ERROR: The type of {key} does not match with the one in Gen3. Skip it"
+                )
+                continue
+            if key == "fdaApproved":
+                if "FDA-approved" in value:
+                    value = "Yes"
+                elif value == "":
+                    value = "Unknown"
+                elif value in ["N/A", "N//A", "N/A*"]:
+                    value = "NA"
+                elif value not in ["Yes", "No", "Unknown", "NA", None]:
+                    value = "Unknown"
+            if key == "customClinicalPhase":
+                if value.lower() == "phase na":
+                    value = "Phase N/A"
+                elif value.lower() in ["preclinical", "pre-clinical"]:
+                    value = "Preclinical Phase"
+                elif value not in [
+                    "Preclinical Phase",
+                    "Phase I",
+                    "Phase I/II",
+                    "Phase II",
+                    "Phase I/II/III",
+                    "Phase III",
+                    "Phase III/IV",
+                    "Phase IV",
+                    "Phase I/III/IV",
+                    "Phase I/IV",
+                    "Phase II/IV",
+                    "Phase II/III/IV",
+                    "Phase I/II/III/IV",
+                    "Phase II/III",
+                    "Phase N/A",
+                    None,
+                ]:
+                    value = None
```

Comment:
why not just use "else"? are we sure about this mapping? sounds to me like these should be mapped to gen3 values, not None

Reply:
if the string value can not be mapped to any enum values, it will be set to None

Reply:
right but i see that these values are in the dictionary. so why are we not submitting them